src/lib/project.js

```diff
@@ -332,7 +339,58 @@ async function scaffoldNext() {
   }
 };`
   );
+  newNextConfig = newNextConfig.replace(
+    'reactStrictMode: true',
+    'reactStrictMode: false'
+  );
   fs.writeFileSync(path.join('ui', 'next.config.js'), newNextConfig);
+
+  const tsconfig = `
+    {
+      "compilerOptions": {
+        "target": "ES2019",
+        "module": "es2022",
+        "lib": ["dom", "dom.iterable", "esnext"],
+        "allowJs": true,
+        "skipLibCheck": true,
+        "strict": true,
+        "strictPropertyInitialization": false,
+        "forceConsistentCasingInFileNames": true,
+        "noEmit": true,
+        "esModuleInterop": true,
+        "module": "esnext",
+        "moduleResolution": "node",
+        "experimentalDecorators": true,
+        "emitDecoratorMetadata": true,
+        "resolveJsonModule": true,
+        "isolatedModules": true,
+        "jsx": "preserve",
+        "incremental": true
+      },
+      "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
+      "exclude": ["node_modules"]
+    }
+  `;
+
+  if (useTypescript == 'yes') {
+    fs.writeFileSync(path.join('ui', 'tsconfig.json'), tsconfig);
+
+    spawnSync(
```

Comment:
`spawnSync` is only needed for interactive CLI commands. 

We should prefer either:
1.  `await shExec('your command');` 
2. Or (my preference), we could do it like [this](https://github.com/o1-labs/zkapp-cli/blob/main/src/lib/project.js#L89-L92). The benefit is that this doesn't require another dependency beyond what we already have. `npm-add-script` appears to be rarely used (1,793 downloads/week), and every additional dep is an additional vector for a supply chain attack, so we could reduce this by just skipping it in this case.

Reply:
agree - updated per (2)